pkg/operator/baremetal_pod.go

```diff
@@ -395,65 +470,62 @@ func createInitContainerStaticIpSet(config *OperatorConfig, baremetalProvisionin
 	return initContainer
 }
 
-func newMetal3Containers(config *OperatorConfig, baremetalProvisioningConfig BaremetalProvisioningConfig) []corev1.Container {
-	//Starting off with the metal3-baremetal-operator container
-	containers := []corev1.Container{
-		{
-			Name:  "metal3-baremetal-operator",
-			Image: config.BaremetalControllers.BaremetalOperator,
-			Ports: []corev1.ContainerPort{
-				{
-					Name:          "metrics",
-					ContainerPort: 60000,
-				},
-			},
-			Command:         []string{"/baremetal-operator"},
-			ImagePullPolicy: "IfNotPresent",
-			VolumeMounts: []corev1.VolumeMount{
-				ironicCredentialsMount,
-				inspectorCredentialsMount,
-			},
-			Env: []corev1.EnvVar{
-				{
-					Name: "WATCH_NAMESPACE",
-					ValueFrom: &corev1.EnvVarSource{
-						FieldRef: &corev1.ObjectFieldSelector{
-							FieldPath: "metadata.namespace",
-						},
+func createContainerMetal3BareMetalOperator(config *OperatorConfig, baremetalProvisioningConfig BaremetalProvisioningConfig) corev1.Container {
+	return corev1.Container{
+		Name:  "metal3-baremetal-operator",
+		Image: config.BaremetalControllers.BaremetalOperator,
+		Command: []string{"/baremetal-operator",
+			"-metrics-addr", metrics.DefaultMetal3MetricsAddress},
+		ImagePullPolicy: "IfNotPresent",
+		Env: []corev1.EnvVar{
```

Comment:
Volume mounts are missing here, which is probably why it is failing to contact ironic.